api/src/main/java/org/apache/cloudstack/api/command/user/vm/AddIpToVmNicCmd.java

```diff
@@ -169,17 +161,15 @@ public long getEntityOwnerId() {
 
     @Override
     public void create() throws ResourceAllocationException {
-        String ip;
         NicSecondaryIp result;
-        String secondaryIp = null;
-        if ((ip = getIpaddress()) != null) {
-            if (!NetUtils.isValidIp4(ip)) {
-                throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, "Invalid ip address " + ip);
-            }
+
+        IpAddresses requestedIpPair = new IpAddresses(ipAddr, null);
+        if (!NetUtils.isIpv4(ipAddr)) {
+            requestedIpPair = new IpAddresses(null, ipAddr);
         }
 
         try {
-            result = _networkService.allocateSecondaryGuestIP(getNicId(), getIpaddress());
+            result = _networkService.allocateSecondaryGuestIP(getNicId(), requestedIpPair);
```

Comment:
What about sending the command to `allocateSecondaryGuestIP`? Then, you can extract the block 166-168 to a method and unit-test it there.